bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,352 @@
+                    for sub_sub_item in sub_item["members"]:
+                        if sub_sub_item["kind"] == "NAMESPACE_REF":
+                            namespace_ref += f'{sub_sub_item["name"]}::'
+                        if sub_sub_item["kind"] == "TYPE_REF":
+                            argument_type_list.append(
+                                f'{namespace_ref}{sub_sub_item["name"]}'
+                            )
+                elif sub_item["element_type"] in ("Float", "Int"):
+                    argument_type_list.append(f'{sub_item["element_type"].lower()}')
+                else:
+                    argument_type_list.append(f'{sub_item["element_type"]}')
+        argument_type_list = ",".join(argument_type_list)
 
-type_functions = {
-    "include": handle_include,
-    "namespace": handle_namespace,
-    "alias": handle_alias,
-    "struct": handle_struct,
-    "constructor": handle_constructor,
-    "operator": handle_operator,
-    "macro": handle_macro,
-}
+        self.linelist.append(f".def(py::init<{argument_type_list}>())")
+
+    def handle_class_template(self):
+        flag = False
+
+        # TODO: Use list based method, like in handle_struct_decl
+        for sub_item in self.members:
+            if sub_item["kind"] == "TEMPLATE_NON_TYPE_PARAMETER":
+                self.linelist.append(
+                    f'template< {sub_item["element_type"].lower()} {sub_item["name"]} >'
+                )
+                flag = True
+        if not flag:
+            self.linelist.append(f"template<>")
+        cxx_base_specifier_list = [
+            sub_item["name"]
+            for sub_item in self.members
+            if sub_item["kind"] == "CXX_BASE_SPECIFIER"
+        ]
+        if cxx_base_specifier_list:
+            cxx_base_specifier_list = ",".join(cxx_base_specifier_list)
+            self.linelist.append(
+                f'py::class_<{self.name, cxx_base_specifier_list}>(m, "{self.name}")'
+            )
+        else:
+            self.linelist.append(f'py::class_<{self.name}>(m, "{self.name}")')
+
+    def handle_inclusion_directive(self):
+        # TODO: update blacklist
+        # blacklist = ["pcl/memory.h", "pcl/pcl_macros.h",]
+        if self.name.startswith("pcl"):
+            self.inclusion_list.append(self.name)
+
+    def handle_final(self, filename, module_name):
+        final = [f"#include <{filename}>"]
+        # TODO: Inclusion list path fix needed
+        # TODO: Currently commented, to be written later
+        # for inclusion in self.inclusion_list:
+        #     final.append(f"#include <{inclusion}>")
+        final += self.pybind_linelist
+        for i in range(len(self.linelist)):
+            if self.linelist[i].startswith("namespace"):
+                continue
+            else:
+                self.linelist[i] = "".join(
+                    (f"PYBIND11_MODULE({module_name}, m)", "{", self.linelist[i])
+                )
+                break
```

Comment:
```suggestion
        for line in self.linelist:
            if line.startswith("namespace"):
                continue
            else:
                line = "".join(
                    (f"PYBIND11_MODULE({module_name}, m)", "{", line)
                )
                break
```

Reply:
I'll address this (along with other function's internal's changes) in another PR.

Reply:
@aponza
We want to change `self._linelist[i]`, not its `line` (strings are immutable -> copy created)

We need index based:
```py
self.linelist[i] = "".join((f"PYBIND11_MODULE({module_name}, m)", "{", self.linelist[i]))
```

Reply:
not sure I got you, still, if you need an index, don't use `for i in range(len(x))` but use `for i, _ in enumerate(x)`

Reply:
Oh yeah, that was really vague what I wrote :joy: 
Better explained via code:
```py
>>> str_list = ["str0", "str1", "str2"]
>>> compare_list = []



>>> for item in str_list:
...     if item == "str1":
...         item = "str100"
...     compare_list.append(item)
... 
>>> print(compare_list)
['str0', 'str100', 'str2']
>>> print(str_list)
['str0', 'str1', 'str2']



>>> for i in range(len(str_list)):
...     if str_list[i] == "str1":
...         str_list[i] = "str100"
... 
>>> print(str_list)
['str0', 'str100', 'str2']
```

I'll do `enumerate` :+1:

Reply:
Now I get what you meant and of course it made sense! Cheers

Reply:
Done.